tests/integration/modules/mac_timezone.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: utf-8 -*-
+'''
+Integration tests for mac_timezone
+'''
+
+# Import python libs
+from __future__ import absolute_import
+from datetime import datetime
+
+# Import Salt Testing libs
+from salttesting.helpers import ensure_in_syspath
+ensure_in_syspath('../../')
+
+# Import salt libs
+import integration
+import salt.utils
+
+USE_NETWORK_TIME = False
+TIME_SERVER = 'time.apple.com'
+TIME_ZONE = ''
+CURRENT_DATE = ''
+CURRENT_TIME = ''
+
+
+class MacTimezoneModuleTest(integration.ModuleCase):
+    '''
+    Validate the mac_timezone module
+    '''
+
+    def setUp(self):
+        '''
+        Get current settings
+        '''
+        if not salt.utils.is_darwin():
+            self.skipTest('Test only available on Mac OS X')
+
+        if not salt.utils.which('systemsetup'):
+            self.skipTest('Test requires systemsetup binary')
+
+        if salt.utils.get_uid(salt.utils.get_user()) != 0:
+            self.skipTest('Test requires root')
+
+        USE_NETWORK_TIME = self.run_function('timezone.get_using_network_time')
+        TIME_SERVER = self.run_function('timezone.get_time_server')
+        TIME_ZONE = self.run_function('timezone.get_zone')
+        CURRENT_DATE = self.run_function('timezone.get_date')
+        CURRENT_TIME = self.run_function('timezone.get_time')
+
+        self.run_function('timezone.set_using_network_time', [False])
+
+    def tearDown(self):
+        '''
+        Reset to original settings
+        '''
+        self.run_function('timezone.set_time_server', [TIME_SERVER])
+        self.run_function('timezone.set_using_network_time', [USE_NETWORK_TIME])
+        self.run_function('timezone.set_zone', [TIME_ZONE])
+        if not USE_NETWORK_TIME:
+            self.run_function('timezone.set_date', [CURRENT_DATE])
+            self.run_function('timezone.set_time', [CURRENT_TIME])
+
+    def test_get_set_date(self):
+        '''
+        Test timezone.get_date
+        Test timezone.set_date
+        '''
+        # Correct Functionality
+        self.assertTrue(self.run_function('timezone.set_date', ['2/20/2011']))
+        self.assertEqual(self.run_function('timezone.get_date'), '2/20/2011')
+
+        # Test bad date format
+        self.assertEqual(
+            self.run_function('timezone.set_date', ['13/12/2014']),
+            'ERROR executing \'timezone.set_date\': '
+            'Invalid Date/Time Format: 13/12/2014'
+        )
+
+    def test_get_set_time(self):
+        '''
+        Test timezone.get_time
+        Test timezone.set_time
+        '''
+        # Correct Functionality
+        self.assertTrue(self.run_function('timezone.set_time', ['3:14']))
```

Comment:
If we're setting timezones here then we need to mark these tests as destructive.